GitCommands/Git/GitModule.cs

```diff
@@ -2542,6 +2542,24 @@ public static bool IsDetachedHead(string branch)
             return DetachedPrefixes.Any(a => branch.StartsWith(a, StringComparison.Ordinal));
         }
 
+        private static Regex detachedHeadRegex = new Regex(@"^\(.* (?<sha1>.*)\)$");
+
+        public static bool TryParseDetachedHead(string text, out string sha1)
+        {
+            sha1 = null;
+            if (IsDetachedHead(text))
```

Comment:
Oh,  sheet! You're right! I forgot it during my last refactoring... I'll fix it.